### DISCUSSION TITLE
PnP Modern results page performance issue

### DISCUSSION THREAD

**Discussion by GangadharMFord** [microsoft-search/pnp-modern-search#2553] [Q&A]
Hi Everyone, 
I have noticed the page performance degradation with VPN (Using Cisco VPN tool -17 seconds) and without VPN (5 seconds) for results page (have all the web parts within the same page).
So, my question here, to improve the page loading performance does we need to allow any specific url's from networking perceptive for faster loading the results page?

**Comment by wobba** [comment]
You should check if you have enabled CDN for SharePoint framework or not. And also verify if it's loading of the web part bundle or search API call which is slower. That should give you a pointer on what to look at.

**Reply by GangadharMFord** [reply]
> You should check if you have enabled CDN for SharePoint framework or not. And also verify if it's loading of the web part bundle or search API call which is slower. That should give you a pointer on what to look at.

Thank you @wobba for your information.
Yes, the CDN were enabled for Search site and I'm seeing the below Urls taking longer time when I'm in VPN connected (the loading time is very low if I'm not connected to VPN). I have also checked at Page performance dignostic for sharepoint tool and below urls causing or taking longer time than expected. please suggest me how to improve the page performance.
Example URls:

https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-us_56fe4162539a079e3407a2c259d53251.js
https://res-1.cdn.office.net/files/odsp-web-prod_2022-11-04.001/spappbarwebpack/spappbarwebpack.js
https://res-1.cdn.office.net/files/sp-client/chunk.SPAppBarCallout_none_cdeda955098896113191.js
https://Tenant URL/sites/site/_layouts/15/online/handlers/SpoSuiteLinks.ashx?Locale=en-US&v=2&mobilereq=0&msajax=1
https://res-1.cdn.office.net/files/sp-client/decorators_none_d89caa363e4fdae45ceb.js
https://res-1.cdn.office.net/files/sp-client/sp-list-webpart_en-us_7213726a888552dace3b.js
https://Tenant url/sites/sites/_api/web/GetClientSideComponents
https://res-1.cdn.office.net/files/sp-client/sp-listview-common_none_b83397603335b14bf82b.js

**Reply by wobba** [reply]
You should reach out to your IT department on why loading cdn URL's are slower over your VPN. It's an infrastructure issue.

**Reply by GangadharMFord** [reply]
> You should reach out to your IT department on why loading cdn URL's are slower over your VPN. It's an infrastructure issue.

Thank you @wobba for your guidance and I'm checking internally with my network team.
